source/common/http/http_server_properties_cache_manager_impl.cc

```diff
@@ -51,9 +51,12 @@ HttpServerPropertiesCacheSharedPtr HttpServerPropertiesCacheManagerImpl::getCach
         factory.createStore(kv_config, data_.validation_visitor_, dispatcher, data_.file_system_);
   }
 
-  HttpServerPropertiesCacheSharedPtr new_cache = std::make_shared<HttpServerPropertiesCacheImpl>(
-      dispatcher, std::move(store), options.max_entries().value());
+  auto new_cache = std::make_shared<HttpServerPropertiesCacheImpl>(dispatcher, std::move(store),
+                                                                   options.max_entries().value());
 
+  for (const std::string& suffix : options.canonical_suffixes()) {
+    new_cache->addCanonicalSuffix(suffix);
```

Comment:
should we have input validation this starts with . and is otherwise a valid hostname?

Reply:
Added the "starts with '.'" check though I'm not sure if we really want to do a valid hostname check. I could imagine some internal usage where folks use "intranet hostnames". But if you feel strongly, happy to reconsider.